File: codex_poting/java/src/main/java/com/example/demo/mybatis/typehandler/BookStockMovementSourceTypeHandler.java
package com.example.demo.mybatis.typehandler;

import com.example.demo.data.domain.BookStockMovementSourceType;
import org.apache.ibatis.type.BaseTypeHandler;
import org.apache.ibatis.type.JdbcType;
import org.apache.ibatis.type.MappedTypes;

import java.sql.CallableStatement;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

@MappedTypes(BookStockMovementSourceType.class)
public class BookStockMovementSourceTypeHandler extends BaseTypeHandler<BookStockMovementSourceType> {
    @Override
    public void setNonNullParameter(PreparedStatement ps, int i, BookStockMovementSourceType parameter, JdbcType jdbcType) throws SQLException {
        ps.setInt(i, parameter.getValue());
    }

    @Override
    public BookStockMovementSourceType getNullableResult(ResultSet rs, String columnName) throws SQLException {
        final var value = rs.getInt(columnName);
        return rs.wasNull() ? null : BookStockMovementSourceType.of(value);
    }

    @Override
    public BookStockMovementSourceType getNullableResult(ResultSet rs, int columnIndex) throws SQLException {
        final var value = rs.getInt(columnIndex);
        return rs.wasNull() ? null : BookStockMovementSourceType.of(value);
    }

    @Override
    public BookStockMovementSourceType getNullableResult(CallableStatement cs, int columnIndex) throws SQLException {
        final var value = cs.getInt(columnIndex);
        return cs.wasNull() ? null : BookStockMovementSourceType.of(value);
    }
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/mybatis/typehandler/BookStockMovementSourceTypeHandler.kt
package com.example.demo.mybatis.typehandler

import com.example.demo.data.domain.BookStockMovementSourceType
import org.apache.ibatis.type.BaseTypeHandler
import org.apache.ibatis.type.JdbcType
import org.apache.ibatis.type.MappedTypes
import java.sql.CallableStatement
import java.sql.PreparedStatement
import java.sql.ResultSet
import java.sql.SQLException

@MappedTypes(BookStockMovementSourceType::class)
class BookStockMovementSourceTypeHandler : BaseTypeHandler<BookStockMovementSourceType?>() {
    @Throws(SQLException::class)
    override fun setNonNullParameter(
        ps: PreparedStatement,
        i: Int,
        parameter: BookStockMovementSourceType?,
        jdbcType: JdbcType?
    ) {
        ps.setInt(i, parameter!!.value)
    }

    @Throws(SQLException::class)
    override fun getNullableResult(rs: ResultSet, columnName: String?): BookStockMovementSourceType? {
        val value = rs.getInt(columnName)
        return if (rs.wasNull()) null else BookStockMovementSourceType.of(value)
    }

    @Throws(SQLException::class)
    override fun getNullableResult(rs: ResultSet, columnIndex: Int): BookStockMovementSourceType? {
        val value = rs.getInt(columnIndex)
        return if (rs.wasNull()) null else BookStockMovementSourceType.of(value)
    }

    @Throws(SQLException::class)
    override fun getNullableResult(cs: CallableStatement, columnIndex: Int): BookStockMovementSourceType? {
        val value = cs.getInt(columnIndex)
        return if (cs.wasNull()) null else BookStockMovementSourceType.of(value)
    }
}
